File: Data Structures & Algorithms/string-encode-and-decode/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @returns {string}
     */
    encode(strs) {
        if(strs.length==0) return ""
        let res=""
       for(let st of strs){
        res+=st.length+"#"+st
       }
       return res
    }

    /**
     * @param {string} str
     * @returns {string[]}
     */
    decode(str) {
        let i=0,res=[]
        while(i<str.length){
            let j=i
            while(str[j]!=="#"){
                j+=1
            }
            let range=parseInt(str.substring(i,j))
            i=j+1
            j=i+range

            let word=str.substring(i,j)
            res.push(word)
            i=j

        }
return res






    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
            
        Lp,Rp=1,1
        n=len(nums)
        res=[1]*n


        for i in range(n):
            
            res[i]*=Lp
            res[n-i-1]*=Rp
            Lp*=nums[i]
            Rp*=nums[n-i-1]
        return res
        